src/domain/pagination/usePagination.tsx

```diff
@@ -23,42 +14,31 @@ type A11yHelpers = {
   resultsLeft: number;
 };
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-type ReturnValue<TData = any, TVariables = OperationVariables> = A11yHelpers & {
-  fetchMore: FetchMoreFunction<TData, TVariables>;
-};
-
-export default function useA11yPagination<
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  TData = any,
-  TVariables = OperationVariables
->({
-  fetchMore: fetchMoreBase,
+export default function useA11yPagination({
   moreResultsAnnouncerRef,
   totalCount,
   pageSize,
   visibleCount,
-}: Options<TData, TVariables>): ReturnValue<TData, TVariables> {
-  const fetchMore = useCallback(
-    async <K extends keyof TVariables>(
-      fetchMoreOptions: FetchMoreQueryOptions<TVariables, K, TData> &
-        FetchMoreOptions<TData, TVariables>
-    ) => {
-      const result = await fetchMoreBase(fetchMoreOptions);
-
-      if (moreResultsAnnouncerRef.current) {
-        // The announcer element is hidden from visual users and as such may
-        // exist in an unnatural place (visually) on the rendered page. If so,
-        // the browser can scroll into an unexpected position. This is true for
-        // instance when the announcer element is absolutely positioned within
-        // a grid.
-        moreResultsAnnouncerRef.current.focus({ preventScroll: true });
-      }
-
-      return result;
-    },
-    [fetchMoreBase, moreResultsAnnouncerRef]
-  );
+}: Options): A11yHelpers {
+  useLayoutEffect(() => {
```

Comment:
I think that using an `effect` here is suspect as we have a clear user action that we can use as a source for this change (user clicking the load more button). Can you open your thinking behind this decision? I would have expected that we can get correct DOM values as we are in an event handler.

Reply:
I need to do little bit thinking on this but I thought layoutEffect needs to be used because then the list has been rerendered (not painted yet) and we can resolve the correct position of the announcer.